### PR TITLE
Goals

### PR DESCRIPTION
**Pull request by frustak**
Implementation of #13 
Added goals section to home page

### Changelog
- Allow user to create goals
- Show list of user goals
- Allow user to delete goals

### Screenshots
image
image
image

### CODE REVIEW

File: src/features/goal/form.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+import { trpc } from "../../utils/trpc";
+import { Button } from "../ui/button";
+
+const schema = z.object({
+  name: z.string().min(1).max(100),
+});
+
+type Schema = z.infer<typeof schema>;
```
Comment (Glyphack) [inline src/features/goal/form.tsx:11]: I did not know you can do this 👏

File: src/features/goal/list.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { Eraser } from "phosphor-react";
+import { trpc } from "../../utils/trpc";
+import { IconButton } from "../ui/button";
+
+type Goal = { id: string; name: string };
+
+export const GoalList = ({ goals }: { goals: Goal[] }) => {
+  if (goals.length === 0)
+    return <p className="text-xs leading-6">Your goal list is empty!</p>;
```
Comment (Glyphack) [inline src/features/goal/list.tsx]: We can write
"How can a man lead his life without any goal?"
😄

**Review by Glyphack — APPROVED** [review]

File: src/features/goal/list.tsx (hunk 2 above)
Comment (frustak) [reply src/features/goal/list.tsx]: Nice suggestion, I'll change it 👍

Comment (frustak) [reply src/features/goal/list.tsx]: image

**Pull request merged**